### PR TITLE
En 973 widgets

### PR DESCRIPTION
**Pull request by fedelocci**
refacturing widgets API

### CODE REVIEW

**Comment by entando-service** [comment]
All is well

File: test/api/widgets.test.js [hunk 1]
```diff
@@ -1,25 +1,99 @@
...
+const correctRequest = {
+  uri: '/api/widgets',
+  method: METHODS.GET,
+  mockResponse: WIDGET_LIST.payload,
+  useAuthentication: true,
+  errors: expect.any(Function),
+};
 
-jest.unmock('api/widget');
+const correctRequestWidget = {
```
Comment (alessioscalici) [inline test/api/widgets.test.js]: You can omit mockResponse and errors (these props are not relevant to the test). In the test, you can use objectContaining, like:

```javascript
expect(makeRequest).toHaveBeenCalledWith(expect.objectContaining({
        uri: `/api/pages/${CONTACTS_PAYLOAD.code}`,
        body: CONTACTS_PAYLOAD,
        method: METHODS.PUT,
        useAuthentication: true,
      }));
```

File: test/state/widgets/actions.test.js [hunk 2]
```diff
@@ -27,47 +28,72 @@ const STATE_MOCKED = {
...
+
+
   it('test SET_STATE_MOCKED for empty object on initial state', () => {
     expect(getWidgetList([])).toEqual(STATE_MOCKED);
   });
 
   it('checks action type', () => {
     const action = getWidgetList();
     expect(action.type).toBe(SET_WIDGET_LIST);
+    expect(action.type).toBe(SET_WIDGET_LIST);
```
Comment (alessioscalici) [inline test/state/widgets/actions.test.js]: Why duplicating this line?

File: test/state/widgets/actions.test.js [hunk 3]
```diff
@@ -27,47 +28,72 @@ const STATE_MOCKED = {
...
-    const store = mockStore(STATE_MOCKED);
-    store.dispatch(fetchWidgetList()).then(() => {
-      const actions = store.getActions();
-      expect(actions[1].payload.widgetList).toBeDefined();
-      done();
+    it('fetchWidgetList is defined and properly valued', (done) => {
+      store.dispatch(fetchWidgetList()).then(() => {
+        expect(getWidgets).toHaveBeenCalled();
+        const actions = store.getActions();
+        const actionPayload = actions[1].payload;
```
Comment (alessioscalici) [inline test/state/widgets/actions.test.js]: Please use toHaveLenght before accessing the array elements, if the test fails the message would be more explicit

File: test/test/mocks/widgets.js [hunk 4]
```diff
@@ -1,3 +1,36 @@
...
+    defaultUi: '<p>Default UI</p>',
+    createdAt: '2018/02/22',
+    updatedAt: '2018/02/22',
+  },
+  errors: [],
+  metaData: {},
+};
+
+export const BODY_ERROR =
+
```
Comment (alessioscalici) [inline test/test/mocks/widgets.js]: Please remove empty line

File: test/test/mocks/widgets.js [hunk 5]
```diff
@@ -1,3 +1,36 @@
+export const BODY_OK =
+{
+  payload: {
```
Comment (alessioscalici) [inline test/test/mocks/widgets.js]: With makeRequest we only need the payload, not the entire response. Please remove payload, errors and metaData level

File: test/test/mocks/widgets.js [hunk 6]
```diff
@@ -1,3 +1,36 @@
...
+    updatedAt: '2018/02/22',
+  },
+  errors: [],
+  metaData: {},
+};
+
+export const BODY_ERROR =
+
+{
+  payload: {},
```
Comment (alessioscalici) [inline test/test/mocks/widgets.js]: With makeRequest we only need the payload, not the entire response. Please remove payload, errors and metaData level

**Comment by entando-service** [comment]
All is well

**Pull request merged**